### PR TITLE
Respect Teammate Name Display

### LINKED ISSUE #52: Respect Teammate Name Display
The is a Mattermost setting called [Teammate Name Display](https://docs.mattermost.com/administration/config-settings.html#teammate-name-display) which "specifies how names are displayed in the user interface." We should respect this when displaying names.

### PR DESCRIPTION
**Pull request by Hanzei**
The best rule I came put with is:

- For author name: Use `GetDisplayName(model.SHOW_NICKNAME_FULLNAME)`
- For users at the end Poll screen:  Use `GetDisplayName(model.SHOW_USERNAME)`
As always this PR includes some refactoring. :wink: 

Fixes #52

### CODE REVIEW

File: server/plugin.go [hunk 1]
```diff
@@ -53,12 +54,22 @@ func (p *MatterpollPlugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r
...
+func (p *MatterpollPlugin) ConvertUserIDToDisplayName(userID string) (string, *model.AppError) {
 	user, err := p.API.GetUser(userID)
 	if err != nil {
 		return "", err
 	}
-	// NOTE: We should better fetch the server config and us this instead of model.SHOW_FULLNAME
-	nameFormat := model.SHOW_FULLNAME
-	return user.GetDisplayName(nameFormat), nil
+	displayName := user.GetDisplayName(*p.ServerConfig.TeamSettings.TeammateNameDisplay)
+	if displayName == user.Username {
```
Comment (kaakaa) [inline server/plugin.go]: When I tested in my local, I may have noticed useful spec of `TeammateNameDisplay` settings.
If server returns `@USERNAME`, client converts `@USERNAME` to the display_name according to `TeammateNameDisplay` settings in AccountSettings. Users can see display names in format users want to see.

So we have better to use `model.SHOW_USERNAME` and prepend `@` always. 

How do you think about this?

Comment (Hanzei) [reply server/plugin.go]: Oh, thank you for the hint. This kind of kills the purpose of this PR and make it do mostly refactoring. I updated my code and the description of this PR.

**Comment by Hanzei** [comment]
This is ready for a second round of review.

File: server/plugin.go (hunk 1 above)
Comment (kaakaa) [reply server/plugin.go]: Thank you for a lot of work and sorry for changing the purpose.
Updated code is LGTM, thanks !

**Pull request merged**